stock_cycle_count/models/stock_inventory.py

```diff
@@ -12,23 +12,27 @@
 class StockInventory(models.Model):
     _inherit = 'stock.inventory'
 
-    @api.one
+    @api.multi
+    @api.depends("state", "line_ids")
     def _compute_inventory_accuracy(self):
-        total_qty = sum(self.line_ids.mapped('theoretical_qty'))
-        abs_discrepancy = sum(self.line_ids.mapped(
-            lambda x: abs(x.discrepancy_qty)))
-        if total_qty:
-            self.inventory_accuracy = PERCENT * (
-                total_qty - abs_discrepancy) / total_qty
-        if not self.line_ids and self.state == 'done':
-            self.inventory_accuracy = 100.0
+        for inv in self:
+            theoretical = sum(inv.line_ids.mapped(
+                lambda x: abs(x.theoretical_qty)))
+            abs_discrepancy = sum(inv.line_ids.mapped(
+                lambda x: abs(x.discrepancy_qty)))
+            if theoretical:
+                inv.inventory_accuracy = max(
+                    PERCENT * (theoretical - abs_discrepancy) / theoretical,
+                    0.0)
+            if not inv.line_ids and inv.state == 'done':
```

Comment:
You can get these records out of the loop (as you based your computation on inv_lines : substract self recordset with records with no inv_lines)

Reply:
if I filter the recordset as you propose the inventories with no lines and in state done will end up with an accuracy of 0.0 (the default), while here I'm setting them to 100.0. Am I missing something?

Reply:
No, I want to say you should compute this in batch before the loop and substract these recordset to enter the loop

Reply:
Ok, I got you. Fixed in https://github.com/OCA/stock-logistics-warehouse/pull/346/commits/d1832362eddedebd13813202aa4709f768991611